clippy_lints/src/const_static_lifetime.rs

```diff
@@ -0,0 +1,56 @@
+use syntax::ast::{Item, ItemKind, TyKind, ExprKind};
+use rustc::lint::{LintPass, EarlyLintPass, LintArray, EarlyContext};
+use utils::span_help_and_lint;
+
+/// **What it does:** Checks for constants with an explicit `'static` lifetime.
+///
+/// **Why is this bad?** Adding `'static` to every reference can create very complicated types.
+///
+/// **Known problems:** None.
+///
+/// **Example:**
+/// ```rust
+///  const FOO: &'static [(&'static str, &'static str, fn(&Bar) -> bool)] = &[..]
+/// ```
+/// This code can be rewritten as
+/// ```rust
+///  const FOO: &[(&str, &str, fn(&Bar) -> bool)] = &[...]
+/// ```
+
+declare_lint! {
+    pub CONST_STATIC_LIFETIME, 
+    Warn,
```

Comment:
I don't know if this should be Warn by default, at least not now. This warns against a syntax that was the only possible syntax for a long time, and lots of projects are using it now. Plus there is little value in updating old code except for "it's possible that way now".

Maybe we should ask for the lint level to be a dynamic parameter to rustc's devs. That way we could have a `minimal_rustc_version` parameter in `clippy.toml`, and only have the lint warn if the minimal version of rustc you want to support is higher that whatever version of rustc added that shortcut.